### ISSUE TITLE
Wrong z-component in trajectories

### ISSUE THREAD

**Issue by chraibi** [JuPedSim/jpscore#252]
In Gitlab by @chraibi on Feb 23, 2017, 15:45 [[origin](https://gitlab.version.fz-juelich.de/jupedsim/jpscore/issues/240)]

## Summary

Pedestrian in 3D geometries have the wrong `z` component.

For example 

a subroom defined by

```
A_x="0" B_y="0" C="20.694"
```

The trajectories will be something like

## Expected behavior

```
<agent ID="1"   x="21.363600"   y="17.103700"   z="26.994000" /> 
```

## Actual behavior

```
<agent ID="1"   x="21.363600"   y="17.103700"   z="27.294000" /> 
```

## Inifile + Geometry file to reproduce bug

[HH_ini.xml](https://gitlab.version.fz-juelich.de/JuPedSim/jpscore/uploads/3adda7028a1af441b34b49ea86c6907c/HH_ini.xml)[HH_geo.xml](/uploads/44d2b6ef426a6b211f19646725d8d291/HH_geo.xml)[HH_routing.xml](/uploads/cfe80bd21684d6987fd9f437425ea9cb/HH_routing.xml)

## Relevant logs, files (inifile and geometry) and/or screenshots

In screenshot peds 1, 2 and 3 are manually modified just to show the problem.
![Screen_Shot_2017-02-23_at_15.39.17](/uploads/040e8b95e951ed846c5fb4b694a1bc45/Screen_Shot_2017-02-23_at_15.39.17.png)

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Feb 23, 2017, 15:48

In this [line](https://gitlab.version.fz-juelich.de/jupedsim/jpscore/blob/develop/IO/IODispatcher.cpp#L110)

we write for `z`

```
ped->GetElevation()+0.3
```

Someone knows why +0.3? :confused:

**Comment by chraibi** [comment]
In Gitlab by @ArneGraf on Mar 1, 2017, 12:34

I would guess, that pedestrians on floor level would maybe be occluded or be flickering, if floor colors are enabled.

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 3, 2017, 21:58

created branch [`240-wrong-z-component-in-trajectories`](https://gitlab.version.fz-juelich.de/jupedsim/jpscore/compare/develop...240-wrong-z-component-in-trajectories)

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 4, 2017, 00:43

mentioned in commit b33413937f4a9648c7414cbe30659883dbfe4b58

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 4, 2017, 01:02

mentioned in merge request !4

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 4, 2017, 01:02

closed via merge request !4

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 4, 2017, 01:02

closed via commit b33413937f4a9648c7414cbe30659883dbfe4b58

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 4, 2017, 01:02

closed via commit 314dbd4a27dd8e7b9200bc075554793b70275919

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Jun 13, 2018, 16:41

removed milestone
